### PR TITLE
chore(ci): snapshot analyses test runs on PR to /api

### PR DESCRIPTION
**Pull request by y3rsh**
## Update Analyses Snapshots Workflow

- [x] Run on PRs to /api
- [x] Open a PR into the PR branch if snapshots would change
- [x] Comment on the PR if a PR is opened

### CODE REVIEW

File: .github/workflows/analyses-snapshot-test.yaml [hunk 1]
```diff
@@ -13,14 +13,21 @@ on:
         default: 'edge'
   schedule:
     - cron:  '26 7 * * *' # 7:26 AM UTC
+  pull_request:
+    paths:
+      - 'api/**'
```
Comment (SyntaxColoring) [inline .github/workflows/analyses-snapshot-test.yaml:18]: Does this need to include `shared-data` too?

File: .github/workflows/analyses-snapshot-test.yaml [hunk 2]
```diff
@@ -70,3 +78,15 @@ jobs:
           branch: 'app-testing/${{ env.TARGET }}-from-${{ env.TEST_SOURCE}}'
           base: ${{ env.TEST_SOURCE}}
 
+    - name: Comment on PR
+      if: failure() && github.event_name == 'pull_request'
+      uses: actions/github-script@v7
+      with:
+        script: |
+          const message = 'A PR has been opened to address analyses snapshot changes. Please review the changes here: https://github.com/${{ github.repository }}/pull/${{ steps.create-pull-request.outputs.pull-request-number }}';
```
Comment (SyntaxColoring) [inline .github/workflows/analyses-snapshot-test.yaml:88]: If a PR requires snapshot changes, would we not want those snapshot changes to be in that same PR?

In other words, do we not want to treat this the same way as any other test failure?

**Review by SyntaxColoring — COMMENTED** [review]
Cool!

File: .github/workflows/analyses-snapshot-test.yaml (hunk 1 above)
Comment (y3rsh) [reply .github/workflows/analyses-snapshot-test.yaml:18]: I was not sure what to include.  Should I add `shared-data/python/**` ? or `shared-data/**`

File: .github/workflows/analyses-snapshot-test.yaml (hunk 2 above)
Comment (y3rsh) [reply .github/workflows/analyses-snapshot-test.yaml:88]: We do.  The PR to update the snapshots will be opened into the  `github.head_ref` branch for the PR.
So you would look at the changes and if they make sense you merge that PR into your branch so it becomes a part of the change into edge.
This code generates a comment on your PR so you have a reminder/link to the PR in addition to the failed "test".

File: .github/workflows/analyses-snapshot-test.yaml (hunk 1 above)
Comment (SyntaxColoring) [reply .github/workflows/analyses-snapshot-test.yaml:18]: Hm, good question. `api-test-lint-deploy.yaml` does:

```
      - 'shared-data/**/*'
      - '!shared-data/js/**'
```

Which seems reasonable to me. Let's go with that.

File: .github/workflows/analyses-snapshot-test.yaml (hunk 2 above)
Comment (SyntaxColoring) [reply .github/workflows/analyses-snapshot-test.yaml:88]: Ohhh interesting. Will my PR get a red ❌ on it if it has an unresolved snapshot diff?

File: .github/workflows/analyses-snapshot-test.yaml (hunk 1 above)
Comment (y3rsh) [inline .github/workflows/analyses-snapshot-test.yaml:18]:
```suggestion
      - 'api/**'
      - 'shared-data/**/*'
      - '!shared-data/js/**'
```

File: .github/workflows/analyses-snapshot-test.yaml (hunk 2 above)
Comment (y3rsh) [reply .github/workflows/analyses-snapshot-test.yaml:88]: Yes it would.

**Review by SyntaxColoring — APPROVED** [review]
Yay!

**Pull request merged**